src/OmniSharp.Abstractions/Models/v1/Diagnostics/DiagnosticLocation.cs

```diff
@@ -1,8 +1,28 @@
-﻿namespace OmniSharp.Models.Diagnostics
+﻿using System.Collections.Generic;
+
+namespace OmniSharp.Models.Diagnostics
 {
     public class DiagnosticLocation : QuickFix
     {
         public string LogLevel { get; set; }
         public string Id { get; set; }
+
+        public override bool Equals(object obj)
+        {
+            var location = obj as DiagnosticLocation;
+            return location != null &&
+                   base.Equals(obj) &&
+                   LogLevel == location.LogLevel &&
+                   Id == location.Id;
+        }
+
+        public override int GetHashCode()
+        {
+            var hashCode = -1670479257;
```

Comment:
have you seen [`System.HashCode.Combine(...)`](https://github.com/dotnet/corefx/issues/14354)? 

Not sure we can use 2.1 runtime in omnisharp but could clean this up a bit if available.

Reply:
Omnisharp targets 4.6.1 and Mono so it is not available 😞